src/luos.c

```diff
@@ -397,62 +468,35 @@ uint8_t Luos_SendMsg(module_t *module, msg_t *msg)
 /******************************************************************************
  * @brief read last msg from buffer for a module
  * @param module who receive the message we are looking for
- * @return received message pointer
+ * @param returned_msg oldest message of the module
+ * @return FAIL if no message available
  ******************************************************************************/
-msg_t *Luos_ReadMsg(module_t *module)
+error_return_t Luos_ReadMsg(module_t *module, msg_t *returned_msg)
 {
-    if (module->message_available > MSG_BUFFER_SIZE)
-    {
-        // msg read too slow
-    }
-    if (module->message_available)
-    {
-        // check if there is a message for this module
-        int i = Mngr_GetNextModuleID(module);
-        if (i >= 0)
-        {
-            // this module have a message, get it
-            mngr_t chunk;
-            Mngr_GetIndexMsg(i, 0, &chunk);
-            return chunk.msg;
-        }
-    }
-    return 0;
+    return MsgAlloc_PullMsg(module->vm, returned_msg);
```

Comment:
We should check if this message is for Luos using Luos_MsgHandler. If it is repull a new message.